Libraries/pbxbuild/Sources/Phase/Context.cpp

```diff
@@ -263,7 +263,7 @@ Group(std::vector<Tool::Input> const &files)
          * the ungrouped  inputs so they don't get added again to the result.
          */
         ungrouped.erase(std::remove_if(ungrouped.begin(), ungrouped.end(), [&](Tool::Input const &ungroupedFile) {
-            if (ungroupedFile.fileType()->identifier() == "text.plist.strings") {
+            if (ungroupedFile.fileType() != nullptr && ungroupedFile.fileType()->identifier() == "text.plist.strings") {
```

Comment:
`pbxspec::PBX::Specification` has to be a pointer since it's `virtual`. And `optional<SomePointer>` is confusing because it could be `nullopt`, `nullptr`, or a value.

Two options: make a non-nullable pointer type and wrap that in `optional`, or check `nullptr` like this does. Neither are great. (Rust gets this right, too...)